sdk/python/toolchain/uv.go

```diff
@@ -213,7 +217,12 @@ func (u *uv) ListPackages(ctx context.Context, transitive bool) ([]PythonPackage
 	}
 	if out, err := cmd.CombinedOutput(); err != nil {
 		if strings.Contains(string(out), "No module named pip") {
-			cmd := exec.CommandContext(ctx, "uvx", "pip", "list", "--format", "json", "-v")
+			pythonInterpreter := filepath.Join(u.virtualenvPath, virtualEnvBinDirName(), "python")
+			args = []string{"pip", "--python", pythonInterpreter, "list", "--format", "json", "-v"}
```

Comment:
We need to set the Python interpreter here, otherwise pip will use the one from the isolated venv used by `uvx`, which will give us the packages from that associated venv, but we need to look at the project's venv instead.